apis/datadoghq/v2alpha1/datadogagent_types.go

```diff
@@ -37,12 +37,375 @@ type DatadogAgentSpec struct {
+	// ClusterChecksRunner configuration.
+	ClusterChecksRunner *ClusterChecksRunnerFeatureConfig `json:"clusterChecksRunner,omitempty"`
+	// PrometheusScrape configuration.
+	PrometheusScrape *PrometheusScrapeFeatureConfig `json:"prometheusScrape,omitempty"`
+	// DatadogMonitor configuration.
+	DatadogMonitor *DatadogMonitorFeatureConfig `json:"datadogMonitor,omitempty"`
+}
+
+// Configuration structs for each feature in DatadogFeatures. All parameters are optional and have default values when necessary.
+// Note: configuration in DatadogAgentSpec.Override takes precedence.
+
+// APMFeatureConfig contains APM (Application Performance Monitoring) configuration.
+// APM runs in the Trace Agent.
+type APMFeatureConfig struct {
+	// Enabled enables Application Performance Monitoring.
+	// Default: false
+	// +optional
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// HostPort takes a port number (0 < x < 65536) to expose on the host. (Most containers do not need this.)
+	// If HostNetwork is enabled, this value must match the ContainerPort.
+	// Default: 8126
+	// +optional
+	HostPort *int32 `json:"hostPort,omitempty"`
+
+	// EnableUnixDomainSocket enables UDS configuration.
+	// See also: https://docs.datadoghq.com/agent/kubernetes/apm/?tab=helm#agent-environment-variables
+	// +optional
+	EnableUnixDomainSocket *bool `json:"enableUnixDomainSocket,omitempty"`
+
+	// SocketPath defines the socket path used when APM over UDS is enabled.
+	// See also: https://docs.datadoghq.com/agent/kubernetes/apm/?tab=helm#agent-environment-variables
+	// Default: `/var/run/datadog/apm.socket`
+	// +optional
+	SocketPath *string `json:"socketPath,omitempty"`
+}
+
+// LogCollectionFeatureConfig contains Logs configuration.
+// Logs collection is run in the Agent.
+type LogCollectionFeatureConfig struct {
+	// Enabled enables Log collection.
+	// Default: false
+	// +optional
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// ContainerCollectAll enables Log collection from all containers.
+	// Default: false
+	// +optional
+	ContainerCollectAll *bool `json:"containerCollectAll,omitempty"`
+
+	// ContainerCollectUsingFiles enables log collection from files in `/var/log/pods instead` of using the container runtime API.
+	// Collecting logs from files is usually the most efficient way of collecting logs.
+	// See also: https://docs.datadoghq.com/agent/basic_agent_usage/kubernetes/#log-collection-setup
+	// Default: true
+	// +optional
+	ContainerCollectUsingFiles *bool `json:"containerCollectUsingFiles,omitempty"`
+
+	// ContainerLogsPath allows log collection from the container log path.
+	// Set to a different path if you are not using the Docker runtime.
+	// See also: https://docs.datadoghq.com/agent/kubernetes/daemonset_setup/?tab=k8sfile#create-manifest
+	// Default: `/var/lib/docker/containers`
+	// +optional
+	ContainerLogsPath *string `json:"containerLogsPath,omitempty"`
```

Comment:
The Env AD does not deal with the logs at the moment. Currently the Logs Agent hardcodes the logs dir to `/var/lib/docker/containers` for the Docker tailer, so if customer uses a custom path it allows to remap.

It could be necessary to any other container runtime **outside Kubernetes**.